rpcs3/Emu/System.cpp

```diff
@@ -2113,15 +2113,30 @@ game_boot_result Emulator::Load(const std::string& title_id, bool is_disc_patch,
 
 				if (from_hdd0_game && m_cat == "DG")
 				{
-					argv[0] = "/dev_bdvd/PS3_GAME/" + unescape(resolved_path.substr(resolved_hdd0.size() + 10));
-					m_dir = "/dev_hdd0/game/" + resolved_path.substr(resolved_hdd0.size(), 10);
+					const std::string tail = resolved_path.substr(resolved_hdd0.size());
+					const std::string tail_usrdir = tail.substr(tail.find_first_of(fs::delim) + 1);
+					const std::string dirname = tail.substr(0, tail.find_first_of(fs::delim));
+					argv[0] = "/dev_bdvd/PS3_GAME/" + unescape(tail_usrdir);
+					m_dir = "/dev_hdd0/game/" + dirname + "/";
 					sys_log.notice("Disc path: %s", m_dir);
+
+					if (dirname != m_title_id)
+					{
+						vfs::mount("/dev_hdd0/game/" + m_title_id, vfs::get("/dev_hdd0/game/" + dirname));
```

Comment:
This cant be correct

Reply:
Do you have proof that the paths in the game are hardcoded to title_id?

Reply:
No. but I found the real issue.

Reply:
> Do you have proof that the paths in the game are hardcoded to title_id?

Yes, tons of totally different SCEEXE000 games did hardcode the `/dev_hdd0/game/title_id` path.
![Image](https://github.com/RPCS3/rpcs3/assets/17809637/642a4891-3beb-4848-9689-416b5af01240)

This PR *was* very helpful for the scenario in which you have tons of such games.